crates/red_knot_python_semantic/src/types.rs

```diff
@@ -292,6 +292,13 @@ impl<'db> Type<'db> {
         matches!(self, Type::Never)
     }
 
+    pub const fn as_tuple_type(&self) -> Option<&TupleType<'db>> {
```

Comment:
I think the reason those method names are using the `into_` prefix is because they consume `self` and return the relevant type while this method only returns a reference.

Also, this method isn't required now but was in an earlier version. Sorry for the churn, I'll remove this.

Reply:
Yes, the `into_*` naming convention is because it consumes `self` -- but because `Type` is `Copy`, consuming `self` and returning an owned version of the underlying data is actually more ergonomic (and probably no less performant, or at least not significantly so) for these methods than returning a reference to the underlying data